test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
Good effort! think about the negative test cases too. If `isMutating` has not been implemented this would pass too so think of other conditions you can capture to display the effects of having `isMutating`